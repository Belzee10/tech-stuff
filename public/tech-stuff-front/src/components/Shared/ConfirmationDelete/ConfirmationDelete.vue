<template>
  <div>
    <p>{{ message }}</p>
    <div class="d-flex justify-end">
      <v-btn
        color="error"
        small
        class="text-capitalize mr-1 confirm-delete"
        @click="() => $emit('confirm-delete')"
      >
        Delete
      </v-btn>
      <v-btn
        small
        class="text-capitalize confirm-cancel"
        @click="() => $emit('confirm-cancel')"
        >Cancel</v-btn
      >
    </div>
  </div>
</template>

<script>
export default {
  name: 'ConfirmationDelete',
  props: {
    message: {
      type: String,
      default: ''
    }
  }
};
</script>

<style lang="scss" scoped></style>
