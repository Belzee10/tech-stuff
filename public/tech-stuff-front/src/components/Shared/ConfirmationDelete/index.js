export { default } from './ConfirmationDelete.vue';
