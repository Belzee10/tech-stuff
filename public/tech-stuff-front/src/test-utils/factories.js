import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
