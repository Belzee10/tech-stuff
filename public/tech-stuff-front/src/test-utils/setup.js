import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
