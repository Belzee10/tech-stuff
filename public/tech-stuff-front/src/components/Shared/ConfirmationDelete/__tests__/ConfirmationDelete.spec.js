import { createWrapper } from '@/test-utils/factories';
import ConfirmationDelete from '../ConfirmationDelete.vue';

describe('ConfirmationDelete.vue', () => {
  test('should ', () => {
    const wrapper = createWrapper(ConfirmationDelete);
    expect(wrapper.element).toMatchSnapshot();
  });
});
