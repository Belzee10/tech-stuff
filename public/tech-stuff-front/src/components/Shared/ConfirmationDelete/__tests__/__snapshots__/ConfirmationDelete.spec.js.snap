// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDelete.vue should  1`] = `
<div>
  <p>
    
  </p>
   
  <div
    class="d-flex justify-end"
  >
    <button
      class="text-capitalize mr-1 confirm-delete v-btn v-btn--contained theme--light v-size--small error"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
      Delete
    
      </span>
    </button>
     
    <button
      class="text-capitalize confirm-cancel v-btn v-btn--contained theme--light v-size--small"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Cancel
      </span>
    </button>
  </div>
</div>
`;
